Firestore/core/test/unit/local/leveldb_index_manager_test.cc

```diff
@@ -38,6 +44,75 @@ INSTANTIATE_TEST_SUITE_P(LevelDbIndexManagerTest,
                          IndexManagerTest,
                          ::testing::Values(PersistenceFactory));
 
+class LevelDbIndexManagerTest : public ::testing::Test {
+ public:
+  // `GetParam()` must return a factory function.
+  LevelDbIndexManagerTest() : persistence{PersistenceFactory()} {
+  }
+
+  std::unique_ptr<Persistence> persistence;
+};
+
+TEST_F(LevelDbIndexManagerTest, CreateReadFieldsIndexes) {
+  persistence->Run("CreateReadDeleteFieldsIndexes", [&]() {
+    IndexManager* index_manager = persistence->index_manager();
+    index_manager->Start();
+
+    index_manager->AddFieldIndex(
+        MakeFieldIndex("coll1", 1, model::FieldIndex::InitialState(), "value",
+                       model::Segment::kAscending));
+    index_manager->AddFieldIndex(
+        MakeFieldIndex("coll2", 2, model::FieldIndex::InitialState(), "value",
+                       model::Segment::kContains));
+
+    {
+      auto indexes = index_manager->GetFieldIndexes("coll1");
+      EXPECT_EQ(indexes.size(), 1);
+      // Note index_id() is 0 because index manager rewrites it using its
+      // internal id.
+      EXPECT_EQ(indexes[0].index_id(), 0);
+      EXPECT_EQ(indexes[0].collection_group(), "coll1");
+    }
+
+    index_manager->AddFieldIndex(
+        MakeFieldIndex("coll1", 3, model::FieldIndex::InitialState(),
+                       "newValue", model::Segment::kContains));
+    {
+      auto indexes = index_manager->GetFieldIndexes("coll1");
+      EXPECT_EQ(indexes.size(), 2);
+      EXPECT_EQ(indexes[0].collection_group(), "coll1");
+      EXPECT_EQ(indexes[1].collection_group(), "coll1");
+    }
+
+    {
+      auto indexes = index_manager->GetFieldIndexes("coll2");
+      EXPECT_EQ(indexes.size(), 1);
+      EXPECT_EQ(indexes[0].collection_group(), "coll2");
+    }
+  });
+}
+
+TEST_F(LevelDbIndexManagerTest, DeleteFieldsIndexeRemovesAllMetadata) {
+  persistence->Run("CreateReadDeleteFieldsIndexes", [&]() {
+    IndexManager* index_manager = persistence->index_manager();
+    index_manager->Start();
+
+    auto index = MakeFieldIndex("coll1", 0, model::FieldIndex::InitialState(),
+                                "value", model::Segment::kAscending);
+    index_manager->AddFieldIndex(index);
+    {
+      auto indexes = index_manager->GetFieldIndexes("coll1");
+      EXPECT_EQ(indexes.size(), 1);
+    }
+
+    index_manager->DeleteFieldIndex(index);
+    {
+      auto indexes = index_manager->GetFieldIndexes("coll1");
+      EXPECT_EQ(indexes.size(), 0);
+    }
+  });
+}
```

Comment:
Please add a test that verifies that indexes are loaded during startup (similar to Android's `testChangeUser()`).
You might also want to implement updateCollectionGroup and nextCollectionGroup to update so you can verify the priority queue - but I am okay with doing this in a follow up.